From 9ced54a8f8ac75e60a31e1623e9f7932e4743446 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 11 Jul 2013 15:54:21 -0500
Subject: [PATCH 2/2] imx6s-wandboard: add as a copy of imx6dl-wandboard

v1: 512MB, fec/usb disabled
v2: u-boot looks for s vs sl: imx6s-wandboard.dtb
v3: merge with for-next branch..

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/Makefile            |    1 +
 arch/arm/boot/dts/imx6s-wandboard.dts |  161 +++++++++++++++++++++++++++++++++
 arch/arm/boot/dts/imx6sl.dtsi         |   11 +++
 3 files changed, 173 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6s-wandboard.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index ce1b3a3..c7c4a91 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -135,6 +135,7 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6q-sbc6x.dtb \
 	imx6q-wandboard.dtb \
 	imx6sl-evk.dtb \
+	imx6s-wandboard.dtb \
 	vf610-twr.dtb
 dtb-$(CONFIG_ARCH_MXS) += imx23-evk.dtb \
 	imx23-olinuxino.dtb \
diff --git a/arch/arm/boot/dts/imx6s-wandboard.dts b/arch/arm/boot/dts/imx6s-wandboard.dts
new file mode 100644
index 0000000..0c94754
--- /dev/null
+++ b/arch/arm/boot/dts/imx6s-wandboard.dts
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+/dts-v1/;
+#include "imx6sl.dtsi"
+
+/ {
+	model = "Wandboard i.MX6 Solo Lite Board";
+	compatible = "wand,imx6s-wandboard", "fsl,imx6sl";
+
+	memory {
+		reg = <0x10000000 0x20000000>; /* 512Mb */
+	};
+
+/*
+ * 	regulators {
+ * 		compatible = "simple-bus";
+ * 
+ * 		reg_2p5v: 2p5v {
+ * 			compatible = "regulator-fixed";
+ * 			regulator-name = "2P5V";
+ * 			regulator-min-microvolt = <2500000>;
+ * 			regulator-max-microvolt = <2500000>;
+ * 			regulator-always-on;
+ * 		};
+ * 
+ * 		reg_3p3v: 3p3v {
+ * 			compatible = "regulator-fixed";
+ * 			regulator-name = "3P3V";
+ * 			regulator-min-microvolt = <3300000>;
+ * 			regulator-max-microvolt = <3300000>;
+ * 			regulator-always-on;
+ * 		};
+ * 	};
+ */
+
+/*
+ * 	sound {
+ * 		compatible = "fsl,imx6s-wandboard-sgtl5000",
+ * 			     "fsl,imx-audio-sgtl5000";
+ * 		model = "imx6s-wandboard-sgtl5000";
+ * 		ssi-controller = <&ssi1>;
+ * 		audio-codec = <&codec>;
+ * 		audio-routing =
+ * 			"MIC_IN", "Mic Jack",
+ * 			"Mic Jack", "Mic Bias",
+ * 			"Headphone Jack", "HP_OUT";
+ * 		mux-int-port = <1>;
+ * 		mux-ext-port = <3>;
+ * 	};
+ */
+};
+
+/*
+ * &audmux {
+ * 	pinctrl-names = "default";
+ * 	pinctrl-0 = <&pinctrl_audmux_2>;
+ * 	status = "okay";
+ * };
+ */
+
+/*
+ * &i2c2 {
+ * 	clock-frequency = <100000>;
+ * 	pinctrl-names = "default";
+ * 	pinctrl-0 = <&pinctrl_i2c2_2>;
+ * 	status = "okay";
+ * 
+ * 	codec: sgtl5000@0a {
+ * 		compatible = "fsl,sgtl5000";
+ * 		reg = <0x0a>;
+ * 		clocks = <&clks 169>;
+ * 		VDDA-supply = <&reg_2p5v>;
+ * 		VDDIO-supply = <&reg_3p3v>;
+ * 	};
+ * };
+ */
+
+/*
+ * &iomuxc {
+ * 	pinctrl-names = "default";
+ * 	pinctrl-0 = <&pinctrl_hog>;
+ * 
+ * 	hog {
+ * 		pinctrl_hog: hoggrp {
+ * 			fsl,pins = <
+ * 				MX6QDL_PAD_GPIO_0__CCM_CLKO1 	0x130b0
+ * 				MX6QDL_PAD_GPIO_2__GPIO1_IO02	0x80000000
+ * 				MX6QDL_PAD_EIM_DA9__GPIO3_IO09	0x80000000
+ * 			>;
+ * 		};
+ * 	};
+ * };
+ */
+
+/*
+ * &fec {
+ * 	pinctrl-names = "default";
+ * 	pinctrl-0 = <&pinctrl_enet_1>;
+ * 	phy-mode = "rgmii";
+ * 	status = "okay";
+ * };
+ */
+
+&ssi1 {
+	fsl,mode = "i2s-slave";
+	status = "okay";
+};
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1_1>;
+	status = "okay";
+};
+
+/*
+ * &uart3 {
+ * 	pinctrl-names = "default";
+ * 	pinctrl-0 = <&pinctrl_uart3_2>;
+ * 	fsl,uart-has-rtscts;
+ * 	status = "okay";
+ * };
+ */
+
+/*
+ * &usbh1 {
+ * 	status = "okay";
+ * };
+ */
+
+/*
+ * &usdhc1 {
+ * 	pinctrl-names = "default";
+ * 	pinctrl-0 = <&pinctrl_usdhc1_2>;
+ * 	cd-gpios = <&gpio1 2 0>;
+ * 	status = "okay";
+ * };
+ */
+
+/*
+ * &usdhc2 {
+ * 	pinctrl-names = "default";
+ * 	pinctrl-0 = <&pinctrl_usdhc2_2>;
+ * 	non-removable;
+ * 	status = "okay";
+ * };
+ */
+
+&usdhc3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc3_2>;
+	cd-gpios = <&gpio3 9 0>;
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6sl.dtsi b/arch/arm/boot/dts/imx6sl.dtsi
index d048a78..18bd813 100644
--- a/arch/arm/boot/dts/imx6sl.dtsi
+++ b/arch/arm/boot/dts/imx6sl.dtsi
@@ -589,6 +589,17 @@
 							MX6SL_PAD_SD3_DAT3__SD3_DATA3 0x17059
 						>;
 					};
+
+					pinctrl_usdhc3_2: usdhc3grp-2 {
+						fsl,pins = <
+							MX6SL_PAD_SD3_CMD__SD3_CMD    0x17059
+							MX6SL_PAD_SD3_CLK__SD3_CLK    0x10059
+							MX6SL_PAD_SD3_DAT0__SD3_DATA0 0x17059
+							MX6SL_PAD_SD3_DAT1__SD3_DATA1 0x17059
+							MX6SL_PAD_SD3_DAT2__SD3_DATA2 0x17059
+							MX6SL_PAD_SD3_DAT3__SD3_DATA3 0x17059
+						>;
+					};
 				};
 			};
 
-- 
1.7.10.4

